lib/logstash/inputs/log4j.rb

```diff
@@ -197,4 +195,25 @@ def build_client_socket
   def add_socket_mixin(socket)
     socket.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
   end
+
+  class Log4jInputStream < java.io.ObjectInputStream
+    ALLOWED_CLASSES = ["org.apache.log4j.spi.LoggingEvent"]
+
+    def initialize(*args)
+      super
+      @class_loader = org.jruby.Ruby.getGlobalRuntime.getJRubyClassLoader 
+    end
+
+    def safety_check(name)
+      raise java.io.InvalidObjectException.new("Object type #{name} is not allowed.") if !ALLOWED_CLASSES.include?(name)
+    end
+
+    def resolveClass(desc)
+      name = desc.getName
+      safety_check(name)
+      return java.lang.Class.forName(name, false, @class_loader)
+    rescue java.lang.ClassNotFoundException => e
+      return super
```

Comment:
Why `return super` here? Shouldn't we just propagate `ClassNotFoundException` to the caller?